src/fifolock.jl

```diff
@@ -0,0 +1,143 @@
+@static if VERSION >= v"1.10"
+
+const LOCKED_BIT = 0b01
+
+"""
+    FIFOLock()
+
+Like Base.ReentrantLock, but with strict FIFO ordering.
```

Comment:
I would explain here about the relative performance tradeoff, including explaining that Base.ReentrantLock is _nearly_ FIFO, but has a performance optimziation for throughput that makes the lock unfair. And how the tradeoff here is for fairness at the expense of that optimization

Reply:
Expanded the docstring.